src/NUnitFramework/framework/Constraints/CollectionOrderedConstraint.cs

```diff
@@ -114,13 +168,33 @@ public CollectionOrderedConstraint By(string propertyName)
         public override string Description
         {
             get 
-            { 
-                string desc = propertyName == null
-                    ? "collection ordered"
-                    : "collection ordered by "+ MsgUtils.FormatValue(propertyName);
+            {
+                string desc;
 
-                if (descending)
-                    desc += ", descending";
+                if (this.properties == null || this.properties.Count == 0)
+                {
+                    desc = "collection ordered";
+
+                    if (descending)
+                        desc += ", descending";
+                }
+                else
+                {
+                    var first = this.properties.ElementAt(0);
+                    desc = "collection ordered by " + MsgUtils.FormatValue(first.Key);
+
+                    if (first.Value)
+                        desc += ", descending";
+
+                    for (int i = 1; i < this.properties.Count; i++)
```

Comment:
Are we guaranteed that all dictionary entries will be returned in the same order we added them? Does this guarantee hold across all platforms? If not, the dictionary should probably be replaced by a list of pairs or a pair of lists.